network-store-client/src/main/java/com/powsybl/network/store/client/BusBreakerViewImpl.java

```diff
@@ -0,0 +1,106 @@
+package com.powsybl.network.store.client;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.*;
+import com.powsybl.network.store.model.Resource;
+import com.powsybl.network.store.model.VoltageLevelAttributes;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class BusBreakerViewImpl implements VoltageLevel.BusBreakerView {
+
+    private final Resource<VoltageLevelAttributes> voltageLevelResource;
+
+    private final NetworkObjectIndex index;
+
+    public BusBreakerViewImpl(Resource<VoltageLevelAttributes> voltageLevelResource, NetworkObjectIndex index) {
+        this.voltageLevelResource = voltageLevelResource;
+        this.index = index;
+    }
+
+    static BusBreakerViewImpl create(Resource<VoltageLevelAttributes> voltageLevelResource, NetworkObjectIndex index) {
+        return new BusBreakerViewImpl(voltageLevelResource, index);
+    }
+
+    @Override
+    public Iterable<Bus> getBuses() {
+        return index.getBuses(voltageLevelResource.getId());
+    }
+
+    @Override
+    public Stream<Bus> getBusStream() {
+        return index.getBuses(voltageLevelResource.getId()).stream();
+    }
+
+    @Override
+    public Bus getBus(String busId) {
+        Optional<Bus> optBus = index.getBus(busId);
+        if (optBus.isPresent() && optBus.get().getVoltageLevel().getId().equals(voltageLevelResource.getId())) {
+            return optBus.get();
+        }
+        return null;
+    }
+
+    @Override
+    public BusAdder newBus() {
+        return new ConfiguredBusAdderImpl(voltageLevelResource, index);
+    }
+
+    @Override
+    public void removeBus(String busId) {
+        throw new UnsupportedOperationException("TODO");
+    }
+
+    @Override
+    public void removeAllBuses() {
+        throw new UnsupportedOperationException("TODO");
+    }
+
+    @Override
+    public Iterable<Switch> getSwitches() {
+        return index.getSwitches();
```

Comment:
What is expected here is switches for the current voltage level not all switches?